### ISSUE TITLE
'defineProps' is not defined in SFC

### ISSUE THREAD

**Issue by herzaso** [vuejs/eslint-plugin-vue#1614]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 
- 7.32.0
- **eslint-plugin-vue version:** 
- 7.16.0
- **Node version:** 
- 14.17.5
- **Operating System:** 
- ubuntu

**Please show your full configuration:**

```json5
{
  "env": {
    "browser": true,
    "es2021": true
  },
  "extends": [
    "plugin:vue/vue3-essential",
    "airbnb-base"
  ],
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": [
    "vue"
  ],
  "rules": {
    "max-len": ["error", { "code": 120 }],
    "import/no-extraneous-dependencies": ["error", {"devDependencies": true}],
    "no-param-reassign": ["error", {"props": true, "ignorePropertyModificationsFor": ["state"]}]
  }
}
```

**What did you do?**	

```vue
<script setup>
defineProps({
  pair: Object,
});
</script>
```

**What did you expect to happen?**
Eslint shouldn't have warned about `defineProps` not being defined

**What actually happened?**

```
20:1  error  'defineProps' is not defined  no-undef
```
**Repository to reproduce this issue**
This was in a private repo

**Comment by ota-meshi** [comment]
Please read faq.
https://eslint.vuejs.org/user-guide/#compiler-macros-such-as-defineprops-and-defineemits-are-warned-by-no-undef-rule
